src/vaults/SectorCrossVault.sol

```diff
@@ -0,0 +1,344 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.8.16;
+
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { BatchedWithdraw } from "./ERC4626/BatchedWithdraw.sol";
+import { ERC4626 } from "./ERC4626/ERC4626.sol";
+import { IXAdapter } from "../interfaces/adapters/IXAdapter.sol";
+import { SocketIntegrator } from "../common/SocketIntegrator.sol";
+
+// import "hardhat/console.sol";
+
+contract SectorCrossVault is BatchedWithdraw, SocketIntegrator {
+	enum msgType {
+		NONE,
+		DEPOSIT,
+		REDEEM,
+		REQUESTREDEEM,
+		REQUESTVALUEOFSHARES,
+		EMERGENCYWITHDRAW
+	}
+
+	struct Vault {
+		uint16 chainId;
+		address adapter;
+		bool allowed;
+	}
+
+	struct Request {
+		uint256 timestamp;
+		uint256 chainId;
+		address vault;
+	}
+
+	struct HarvestLedger {
+		uint256 depositValue;
+		bool isOpen;
+		uint256 openIndex;
+		Request[] request;
+	}
+
+	// TODO Implement functions with harvestLock modifier
+
+	// Controls deposits
+	mapping(address => Vault) public depositedVaults;
+
+	// Keep track of depositedVaults
+	// TODO Has to implement push to this array
+	address[] internal vaultsArr;
+
+	// Harvest state
+	HarvestLedger public harvestLedger;
+	// Controls emergency withdraw
+	bool internal emergencyEnabled = false;
```

Comment:
I need to give this some more thought, but we probably don't need this flag.
My latest thinking is to allow emergencyWithdraw if there it has been longer than x time from the last withdraw timestamp. Don't worry too much about this - this is something I can finetune.